From d8cfb89b7db3ee9280e686dbf9018f9d70943624 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Mon, 24 Mar 2025 15:48:11 +0100
Subject: [PATCH] media: dcmipp: add stm32mp21 soc support

Add a new compatible support for the stm32mp21 which similar
to the stm32mp25 except that it also embed a new histogram
block.

Change-Id: I885d89c992fb9d6f475a70ae1320da55a79a85d1
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/446244
Reviewed-by: Alain VOLMAT <alain.volmat@st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Tested-by: Alain VOLMAT <alain.volmat@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../platform/st/stm32/stm32-dcmipp/dcmipp-core.c  | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-core.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-core.c
@@ -184,6 +184,20 @@ static const struct dcmipp_pipeline_conf
 	.pipe_nb	= 3
 };
 
+#define DCMIPP_STM32MP21_VERR  0x31
+static const struct dcmipp_pipeline_config stm32mp21_pipe_cfg = {
+	.ents		= stm32mp25_ent_config,
+	.num_ents	= ARRAY_SIZE(stm32mp25_ent_config),
+	.links		= stm32mp25_ent_links,
+	.num_links	= ARRAY_SIZE(stm32mp25_ent_links),
+	.hw_revision	= DCMIPP_STM32MP21_VERR,
+	.has_csi2	= true,
+	.has_tpg	= true,
+	.has_histo	= true,
+	.needs_mclk	= true,
+	.pipe_nb	= 3
+};
+
 #define LINK_FLAG_TO_STR(f) ((f) == 0 ? "" :\
 			     (f) == MEDIA_LNK_FL_ENABLED ? "ENABLED" :\
 			     (f) == MEDIA_LNK_FL_IMMUTABLE ? "IMMUTABLE" :\
@@ -262,6 +276,7 @@ err_init_entity:
 
 static const struct of_device_id dcmipp_of_match[] = {
 	{ .compatible = "st,stm32mp13-dcmipp", .data = &stm32mp13_pipe_cfg },
+	{ .compatible = "st,stm32mp21-dcmipp", .data = &stm32mp21_pipe_cfg },
 	{ .compatible = "st,stm32mp25-dcmipp", .data = &stm32mp25_pipe_cfg },
 	{ /* end node */ },
 };
